test/dashboard-menu.test.js

```diff
@@ -0,0 +1,33 @@
+import { html, fixture, expect } from '@open-wc/testing';
+import Sinon, { stub } from 'sinon';
+import { Router } from '@vaadin/router';
+import '../src/dashboard/Dashboard-menu.js';
+
+describe('Dashboard Menu details', () => {
+  // Write test cases inside this block
+  // refer basic-details.js files
+  let el;
+  before( async() => {
+    el = await fixture(html `<dashboard-menu><dashboard-menu>`);
+  });
+  
+  it('is accessible', () => {
+    expect(el).to.be.accessible;
+  });
+
+  it('_setTypeInLS() method called ', () => {
+    const spy = Sinon.spy(el, 'navigateToDetails');
+    el._setTypeInLS();
+    el.navigateToDetails();
```

Comment:
why calling navigateToDetails() yourself?.. then of course spy will be called... just call _setTypeInLS() & expect navigateToDetails() to be called. Like this: 
```js
  it('_setTypeInLS() method calls navigateToDetails()', () => {
    const spy = Sinon.spy(el, 'navigateToDetails');
    el._setTypeInLS();
    expect(spy).to.have.called;
    spy.restore();
  });
```

Think what function does.. & test that.